persist/fs/seek.go

```diff
@@ -181,6 +181,13 @@ func (s *seeker) Open(namespace ts.ID, shard uint32, blockStart time.Time) error
 		return err
 	}
 
+	if !s.keepUnreadBuf {
```

Comment:
Ah I see, yeah I misread it and thought somehow this was called before the `readIndex` call. It's probably okay to leave it since both `readInfo` and `readIndex` are unexported methods but I'm fine with moving it to `Open`, too